qiskit/compiler/assemble.py

```diff
@@ -176,6 +174,59 @@ def _parse_run_args(backend, qobj_id, qobj_header,
     them in case a run option is passed through multiple args (explicitly setting an arg
     has more priority than the arg set by backend)
 
+    Returns:
+        RunConfig: a run config, which is a standardized object that configures the qobj
+            and determines the runtime environment.
+    """
+    # grab relevant info from backend if it exists
+    backend_config = None
+    if backend:
+        backend_config = backend.configuration()
+
+    memory_slots = memory_slots or getattr(backend_config, 'memory_slots', None)
+    rep_time = rep_time or getattr(backend_config, 'rep_times', None)
+    if isinstance(rep_time, list):
+        rep_time = rep_time[0]
+
+    # an identifier for the Qobj
+    qobj_id = qobj_id or str(uuid.uuid4())
+
+    # The header that goes at the top of the Qobj (and later Result)
+    # we process it as dict, then write entries that are not None to a QobjHeader object
+    qobj_header = qobj_header or {}
+    if isinstance(qobj_header, QobjHeader):
+        qobj_header = qobj_header.to_dict()
+    backend_name = getattr(backend_config, 'backend_name', None)
+    backend_version = getattr(backend_config, 'backend_version', None)
+    qobj_header = {**dict(backend_name=backend_name, backend_version=backend_version),
+                   **qobj_header}
+    qobj_header = QobjHeader(**{k: v for k, v in qobj_header.items() if v is not None})
+
+    # create run configuration and populate
+    run_config_dict = dict(shots=shots,
+                           memory=memory,
+                           max_credits=max_credits,
+                           seed_simulator=seed_simulator,
+                           memory_slots=memory_slots,
+                           memory_slot_size=memory_slot_size,
+                           rep_time=rep_time,
+                           **run_config)
+
+    return qobj_id, qobj_header, run_config_dict
+
+
+def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
+                      meas_lo_range, schedule_los, meas_level,
+                      meas_return, meas_map, **run_config):
+    """Resolve the various types of args allowed to the assemble() function through
```

Comment:
May be this could be short and sweet, something like.

```Build a pulse RunConfig replacing unset arguments with defaults derived from the `backend`. See `assemble` for more information on the required arguments.```

Same could be said for the circuit parameter extraction.